ganttproject/src/main/java/net/sourceforge/ganttproject/GanttProject.java

```diff
@@ -148,15 +146,36 @@ public class GanttProject extends GanttProjectBase implements ResourceView, Gant
 
   private FXSearchUi mySearchUi;
 
-  /**
-   * Number of transactions committed by the Colloboque server.
-   */
-  private int myCommittedTxnNum = 0;
-  /**
-   * ID of the last transaction committed by the Colloboque server.
-   */
-  private String myLastCommittedTxnId = "";
-  private final Lock myXlogSendingLock = new ReentrantLock();
+  private static class TxnCommitInfo {
```

Comment:
In my opinion,Pair<String, Int> would suffice. Also, it looks as if this class was supposed to be immutable, but something went wrong, namely, that TxnCommitInfoSync which requires a default constructor and non-final members.

I suggest to keep the Sync class, but remove the inheritance and optionally replace TxnCommitInfo with a Pair